walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -268,7 +270,155 @@ public void notEnoughRoom() {
     Collection<TimeRange> expected = Arrays.asList();
 
+    Assert.assertEquals(expected, actual);
+  } 
+
+  @Test
+  public void notEnoughRoomForOptional() {
+    // Based on justEnoughRoom, add an optional attendee B who has an event between 8:30 and 8:45. 
+    // The optional attendee should be ignored since considering their schedule 
+    // would result in a time slot smaller than the requested time.
+    //
+    // Events  : |--A--|     |----A----|
+    //                 |--B-|
+    // Day     : |---------------------|
+    // Options :       |-----|
+    
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TimeRange.END_OF_DAY, true),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 3", TimeRange.fromStartDuration(TIME_0830AM, DURATION_15_MINUTES),
+            Arrays.asList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES));
+
+    Assert.assertEquals(expected, actual);
+  }
+ 
+  @Test
+  public void optionalAttendeesAvailable() {
+
+    // No mandatory attendees, just two optional attendees with several gaps in 
+    // their schedules. Those gaps should be identified and returned.
+    //
+    // Events  : |--A---|
+    //                       |--B--|
+    // Day     : |---------------------|
+    // Options :        |----|     |---|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_1000AM, TIME_1100AM, false),
+            Arrays.asList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(NO_ATTENDEES, DURATION_60_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected = Arrays.asList(TimeRange.fromStartEnd(TIME_0830AM, TIME_1000AM, false),
+            TimeRange.fromStartEnd(TIME_1100AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void optionalAttendeesNotAvailable() {
+
+    // No mandatory attendees, just two optional attendees with no gaps in their schedules. 
+    // query should return that no time is available.
+    //
+    // Events  : |--A------|
+    //                     |----B------|
+    // Day     : |---------------------|
+    // Options :        |----|     |---|
```

Comment:
I think this set of options is incorrect, right? I'm guessing this is a copy paste error.

Reply:
Yep, thanks for catching it!